subsys/bluetooth/controller/ll_sw/ull_iso.c

```diff
@@ -496,13 +519,205 @@ int ull_iso_reset(void)
 	return 0;
 }
 
+#if defined(CONFIG_BT_CTLR_SYNC_ISO) || defined(CONFIG_BT_CTLR_CONN_ISO)
+void *ull_iso_pdu_rx_alloc_peek(uint8_t count)
+{
+	if (count > MFIFO_AVAIL_COUNT_GET(iso_rx)) {
+		return NULL;
+	}
+
+	return MFIFO_DEQUEUE_PEEK(iso_rx);
+}
+
+void *ull_iso_pdu_rx_alloc(void)
+{
+	return MFIFO_DEQUEUE(iso_rx);
+}
+
+void ull_iso_rx_put(memq_link_t *link, void *rx)
+{
+	/* Enqueue the Rx object */
+	memq_enqueue(link, rx, &memq_ull_iso_rx.tail);
+}
+
+void ull_iso_rx_sched(void)
+{
+	static memq_link_t link;
+	static struct mayfly mfy = {0, 0, &link, NULL, iso_rx_demux};
+
+	/* Kick the ULL (using the mayfly, tailchain it) */
+	mayfly_enqueue(TICKER_USER_ID_LLL, TICKER_USER_ID_ULL_HIGH, 1, &mfy);
+}
+
+static void iso_rx_demux(void *param)
+{
+	struct node_rx_hdr *rx;
+	memq_link_t *link;
+
+	do {
+		link = memq_peek(memq_ull_iso_rx.head, memq_ull_iso_rx.tail,
+				 (void **)&rx);
+		if (link) {
+			/* Demux Rx objects */
+			switch (rx->type) {
+			case NODE_RX_TYPE_RELEASE:
+				memq_dequeue(memq_ull_iso_rx.tail, &memq_ull_iso_rx.head, NULL);
+				ll_iso_rx_put(link, rx);
+				ll_rx_sched();
+				break;
+
+			case NODE_RX_TYPE_ISO_PDU:
+				/* Remove from receive-queue; ULL has received this now */
+				(void)memq_dequeue(memq_ull_iso_rx.tail, &memq_ull_iso_rx.head,
+						   NULL);
+
+				struct node_rx_pdu *rx_pdu = (struct node_rx_pdu *)rx;
+				struct ll_conn_iso_stream *cis =
+					ll_conn_iso_stream_get(rx_pdu->hdr.handle);
+				struct ll_iso_datapath *dp = cis->datapath_out;
+				isoal_sink_handle_t sink = dp->sink_hdl;
+
+				if (dp->path_id != BT_HCI_DATAPATH_ID_HCI) {
+					/* If vendor specific datapath pass to ISO AL here,
+					 * in case of HCI destination it will be passed in
+					 * HCI context.
+					 */
+					struct isoal_pdu_rx pckt_meta = {
+						.meta = &rx_pdu->hdr.rx_iso_meta,
+						.pdu  = (union isoal_pdu *) &rx_pdu->pdu[0]
+					};
+
+					/* Pass the ISO PDU through ISO-AL */
+					isoal_status_t err =
+						isoal_rx_pdu_recombine(sink, &pckt_meta);
+
+					LL_ASSERT(err == ISOAL_STATUS_OK); /* TODO handle err */
+				}
+
+				/* Let ISO PDU start its long journey upwards */
+				ll_iso_rx_put(link, rx);
+				ll_rx_sched();
+				break;
+
+			default:
+				LL_ASSERT(0);
+				break;
+			}
+		}
+	} while (link);
+}
+
+void ll_iso_rx_put(memq_link_t *link, void *rx)
+{
+	/* Enqueue the Rx object */
+	memq_enqueue(link, rx, &memq_ll_iso_rx.tail);
+}
+
+void *ll_iso_rx_get(void)
+{
+	struct node_rx_hdr *rx;
+	memq_link_t *link;
+
+	link = memq_peek(memq_ll_iso_rx.head, memq_ll_iso_rx.tail, (void **)&rx);
+	while (link) {
+		/* Do not send up buffers to Host thread that are
+		 * marked for release
+		 */
+		if (rx->type == NODE_RX_TYPE_RELEASE) {
+			(void)memq_dequeue(memq_ll_iso_rx.tail,
+						&memq_ll_iso_rx.head, NULL);
+			mem_release(link, &mem_link_iso_rx.free);
+
```

Comment:
why all those extra blank lines?

Reply:
I guess line 630 and 632 may safely be removed without cluttering.